src/algebra/cubic_discriminant.lean

```diff
@@ -0,0 +1,277 @@
+/-
+Copyright (c) 2022 David Kurniadi Angdinata. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Kurniadi Angdinata
+-/
+
+import field_theory.splitting_field
+
+/-!
+# Cubics and discriminants
+
+This file defines cubic polynomials over a semiring and their discriminants over a splitting field.
+
+## Main definitions
+
+* `cubic`
+* `disc`
+
+## Main statements
+
+* `disc_ne_zero_iff_roots_nodup`
+
+## References
+
+* <https://en.wikipedia.org/wiki/Cubic_equation>
+* <https://en.wikipedia.org/wiki/Discriminant>
```

Comment:
I'm not 100% sure, but I think these will just be linkified without the `<>` brackets.